tools/compiletest/src/runtest.rs

```diff
@@ -310,6 +324,66 @@ impl<'test> TestCx<'test> {
         self.compose_and_run(kani)
     }
 
+    /// Runs an executable file and:
+    ///  * Checks the expected output if an expected file is specified
+    ///  * Checks the exit code (assumed to be 0 by default)
+    fn run_exec_test(&self) {
+        // Open the `config.yml` file and extract its values
+        let path_yml = self.testpaths.file.join("config.yml");
+        let config_file = std::fs::File::open(path_yml).expect("couldn't open `config.yml`");
+        let exec_config_res = serde_yaml::from_reader(config_file);
+        if let Err(error) = &exec_config_res {
+            let err_msg = format!("couldn't parse `config.yml` file: {error}");
+            self.error(&err_msg);
```

Comment:
Will this continue execution after the error? If so, then it would be better to use `fatal_proc_rec` so that this doesn't go unnoticed.

Reply:
You're right, it continues execution. But we couldn't use `fatal_proc_rec` because one of the arguments is `proc_res` which is the process we spawn to run the test. So I used the `fatal_error` function added in the previous revision.

With this change, the output (if I remove `config.yml` for one test) shows like this:
```
running 2 tests
F.
failures:

---- [exec] script-based-init/error-code stdout ----
error: test failed: couldn't find script in /home/ubuntu/kani/tests/script-based-init/error-code/exit-one.sh
```